Spigot-Server-Patches/0648-Add-Zstd-LZ4-support-for-region-files.patch

```diff
@@ -0,0 +1,956 @@
++    // Paper end
+         Main.LOGGER.info("Forcing world upgrade! {}", convertable_conversionsession.getLevelName()); // CraftBukkit
+-        WorldUpgrader worldupgrader = new WorldUpgrader(convertable_conversionsession, datafixer, immutableset, flag);
++        WorldUpgrader worldupgrader = new WorldUpgrader(convertable_conversionsession, datafixer, immutableset, flag, forceWrite); // Paper - add forceWrite flag
+         IChatBaseComponent ichatbasecomponent = null;
+ 
+         while (!worldupgrader.b()) {
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index a7cd98a7cd7fed95e37a178f732339e204650b10..a561fdbc047d6bb39246363fe1b1ecf1f49baf41 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -390,7 +390,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+                     return true;
+                 }, worlddata.getGeneratorSettings().d().d().stream().map((entry) -> {
+                     return ResourceKey.a(IRegistry.K, ((ResourceKey) entry.getKey()).a());
+-                }).collect(ImmutableSet.toImmutableSet()));
++                }).collect(ImmutableSet.toImmutableSet()), options.has("forceWrite")); // Paper - add forceWrite flag
+             }
+ 
+             IWorldDataServer iworlddataserver = worlddata;
+diff --git a/src/main/java/net/minecraft/server/RegionFile.java b/src/main/java/net/minecraft/server/RegionFile.java
+index 1751fb6934d9242e475c1a44b2a4a1ade6987766..c51c6ad766d67160d4def9ea10cadf8f9cf9daca 100644
+--- a/src/main/java/net/minecraft/server/RegionFile.java
++++ b/src/main/java/net/minecraft/server/RegionFile.java
+@@ -65,7 +65,7 @@ public class RegionFile implements AutoCloseable {
+     // Paper end
+ 
+     public RegionFile(File file, File file1, boolean flag) throws IOException {
+-        this(file.toPath(), file1.toPath(), RegionFileCompression.b, flag);
++        this(file.toPath(), file1.toPath(), RegionFileCompression.getDefaultCompressionMethod(), flag); // Paper - new compression methods
+     }
+ 
+     public RegionFile(java.nio.file.Path java_nio_file_path, java.nio.file.Path java_nio_file_path1, RegionFileCompression regionfilecompression, boolean flag) throws IOException {
+diff --git a/src/main/java/net/minecraft/server/RegionFileCompression.java b/src/main/java/net/minecraft/server/RegionFileCompression.java
+index 3382d678e68e559b8d3cb9dced4fce24206cd38f..3ac8bf12ff3dc1189f02a6b3b34cb35a68ce2d21 100644
+--- a/src/main/java/net/minecraft/server/RegionFileCompression.java
++++ b/src/main/java/net/minecraft/server/RegionFileCompression.java
+@@ -1,30 +1,122 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.PaperConfig; // Paper
```

Comment:
group all these imports in one paper block

Reply:
can also add more imports then, as then the amount doesnt really change anything in regards to maintiance burden